src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -1523,6 +1527,68 @@ def process_access_statements(nodes):
 
         return (default_visibility, visibility_map)
 
+    @staticmethod
+    def process_save_statements(nodes, symbol_table):
+        '''
+        Search the supplied list of fparser2 nodes (which must represent a
+        complete Specification Part) for any SAVE statements (e.g.
+        "SAVE :: my_var") to determine which Symbols are static.
+
+        :param nodes: nodes in the fparser2 parse tree describing a
+                      Specification Part that will be searched.
+        :type nodes: List[:py:class:`fparser.two.utils.Base`]
+        :param symbol_table: the table to which to add any UnknownFortranType
+            Symbols required to capture SAVEs of named common blocks.
+        :type symbol_table: :py:class:`psyclone.psyir.symbols.SymbolTable`
+
+        :returns: names of symbols that are static or just "*" if they all are.
+        :rtype: List[str]
+
+        :raises GenerationError: if the parse tree is found to contain a SAVE
+            without a saved-entity list *and* one or more SAVE attributes or
+            SAVE statements (C580).
+
+        '''
+        default_save = False
+        # Set holding the names of those symbols which are marked as static
+        # via an explicit SAVE stmt (e.g. "SAVE :: my_var")
+        explicit_save = set()
+
+        save_stmts = walk(nodes, Fortran2003.Save_Stmt)
+
+        for stmt in save_stmts:
+
+            if not stmt.children[1]:
+                # No saved-entity list means that all entities are static.
+                default_save = True
+            else:
+                symbol_names = [child.string.lower() for child in
+                                stmt.children[1].children]
+                explicit_save.update(symbol_names)
+
+        if default_save:
+            if explicit_save:
+                # This should really be caught by the Fortran parser but
+                # fparser2 is lax.
+                names = sorted(list(explicit_save))
+                raise GenerationError(
+                    f"Supplied nodes contain a SAVE without a saved-entity "
+                    f"list plus one or more SAVES *with* saved-entity lists "
+                    f"(naming {names}). This is not valid Fortran.")
+            explicit_save.add("*")
+
+        # If there are any named Common blocks listed in a SAVE statement then
+        # we create Symbols of UnknownFortranType for them (so that the
+        # backend can recreate the necessary SAVE statement) and remove them
+        # from the list returned by this method.
+        for name in explicit_save.copy():
+            if name.startswith("/"):
+                uftype = UnknownFortranType(f"SAVE :: {name}")
+                symbol_table.new_symbol(name, symbol_type=DataSymbol,
+                                        datatype=uftype)
```

Comment:
It is handled by the try...catch because Fortran does not permit names beginning with underscore. I've followed your suggestion now and removed the try...except.